nomenclature/region_mapping_models.py

```diff
@@ -188,3 +197,34 @@ def validate_mapping(cls, v, values):
         if invalid:
             raise RegionNotDefinedError(invalid, v.file)
         return v
+
+    def apply(self, df: IamDataFrame) -> IamDataFrame:
+        all_dfs: List[IamDataFrame] = []
+        for model in df.model:
+            model_df = df.filter(model=model)
+
+            # If no mapping is defined the data frame is returned unchanged
+            if model not in self.mappings:
+                all_dfs.append(model_df)
+            # Otherwise we first rename, then aggregate
+            else:
+                cm = self.mappings[model]
+
+                # Rename
+                if cm.native_regions is not None:
+                    all_dfs.append(
+                        model_df.rename(region=cm.renaming_dict).filter(
+                            region=[x.target_native_region for x in cm.native_regions]
+                        )
+                    )
+
+                # Aggregate
+                if cm.common_regions is not None:
+                    for cr in cm.common_regions:
+                        agg_df = model_df.aggregate_region(
+                            model_df.variable,
+                            cr.name,
+                            [r.name for r in cr.constituent_regions],
```

Comment:
I was looking at this line for quite a while, thinking why `constituent_regions` has to be iterated over... After some digging, I realized that `CommonRegion.constituent_regions` is a list of `NativeRegion` instances - but why is that necessary? Would a list of strings be sufficient - and make the code a lot more readable?

Reply:
You're right, it is not strictly necessary I made that design choice on more of a conceptual basis to emphasis the fact that a common region is comprised of one or more model native regions.
I can change it to just a list of strings if you want.
Alternatively, I could also add a property `constituent_region_names` or something along those lines to `RegionAggregationMapping`. This way we could move the list comprehension out of the apply function.

Reply:
Still a bit convoluted, but fine to add a new attribute as you suggest.

Reply:
Now that I think about it just a list of strings is probably the better way to go since I don't use the constituent regions for anything else other than selecting model native regions.
I would vote for simplifying it, we can always revert back to using the full `NativeRegion` model if need be.